napari/layers/points/_tests/test_points_mouse_bindings.py

```diff
@@ -841,3 +844,56 @@ def test_drag_start_selection(
         assert False, 'Unreachable code'  # pragma: no cover
     assert layer._drag_box is None
     assert layer._drag_start is None
+
+
+def test_drag_point_with_mouse(create_known_points_layer_2d):
+    layer, n_points, _ = create_known_points_layer_2d
+    layer.events.data = MagicMock()
+    layer.mode = 'select'
+    old_data = (
+        layer.data.copy()
+    )  # ensure you have old data, not updated in place
+    layer.selected_data = {1}
+    initial_position = tuple(layer.data[1])
+
+    new_position = [0, 0]
+    modifier = []
+
+    event = read_only_event(
+        type='mouse_press', position=initial_position, modifiers=modifier
+    )
+    mouse_press_callbacks(layer, event)
+
+    # Required to assert before the changing event as otherwise layer.data for changing is updated in place.
+    changing_event = {
+        "value": old_data,
+        "action": ActionType.CHANGING,
+        "data_indices": (1,),
+        "vertex_indices": ((),),
+    }
+
+    def side_effect(*args, **kwargs):
+        if kwargs["action"] == ActionType.CHANGING:
+            assert compare_dicts(kwargs, changing_event)
+
+    layer.events.data.side_effect = side_effect
+    event = read_only_event(
+        type='mouse_move',
+        is_dragging=True,
+        position=new_position,
+        modifiers=modifier,
+    )
+    mouse_move_callbacks(layer, event)
+
+    event = read_only_event(
+        type='mouse_release', is_dragging=False, modifiers=modifier
+    )
+    mouse_release_callbacks(layer, event)
+
+    changed_event = {
+        "value": layer.data,
+        "action": ActionType.CHANGED,
+        "data_indices": (1,),
+        "vertex_indices": ((),),
+    }
+    assert compare_dicts(layer.events.data.call_args[1], changed_event)
```

Comment:
```suggestion
    assert not np.array_equal(layer.data, old_data)
    assert compare_dicts(layer.events.data.call_args[1], changed_event)
```

To validate if an array is changed (or if there is a separate test for this)